### PR TITLE
Connection manager alias and uri

### PR DESCRIPTION
**Pull request by Caos-Maker89**

### CODE REVIEW

File: src/MongoDBLibrary/mongo_connection_manager.py [hunk 1]
```diff
@@ -12,11 +16,33 @@ def __init__(self):
...
+         Overlay _cache.switch using dictionary
+         Get from cache the dictionary contain dbconnection and api_module
+         and return them
+        """
+        obj_dict = self._cache.switch(alias)
+        db_connection = obj_dict['connection']
+
+        return db_connection
+
+    def connect_to_mongodb(self, alias, dbHost='localhost', dbPort=27017, dbMaxPoolSize=10, dbNetworkTimeout=None,
```
Comment (fprem) [inline src/MongoDBLibrary/mongo_connection_manager.py]: it is better to put the new Alias ​​field last with default set to none

**Review by fprem — APPROVED** [review]

File: src/MongoDBLibrary/mongo_connection_manager.py (hunk 1 above)
Comment (Caos-Maker89) [reply src/MongoDBLibrary/mongo_connection_manager.py]: done

**Review by gabriele-gozzoli — APPROVED** [review]

File: src/MongoDBLibrary/mongo_connection_manager.py [hunk 2]
```diff
@@ -12,11 +16,33 @@ def __init__(self):
         """
         Initializes _dbconnection to None.
         """
-        self._dbconnection = None
+        self._cache = robot.utils.ConnectionCache('No connection created')
         self._builtin = BuiltIn()
 
+    def _push_cache(self, alias=None, connection=None):
+        """
+         Overlay _cache.register using dictionary
```
Comment (lucagiove) [inline src/MongoDBLibrary/mongo_connection_manager.py:24]: Overlay? Maybe override?

File: src/MongoDBLibrary/mongo_connection_manager.py [hunk 3]
```diff
@@ -12,11 +16,33 @@ def __init__(self):
...
+         Create a dictionary that contains the dbconnection and the api_module used
+         and push it into the cache
+        """
+
+        obj_dict = {'connection': connection}
```
Comment (lucagiove) [inline src/MongoDBLibrary/mongo_connection_manager.py:29]: Not needed a dictionary to store just the connection.

File: src/MongoDBLibrary/mongo_connection_manager.py [hunk 4]
```diff
@@ -12,11 +16,33 @@ def __init__(self):
...
         Initializes _dbconnection to None.
         """
-        self._dbconnection = None
+        self._cache = robot.utils.ConnectionCache('No connection created')
         self._builtin = BuiltIn()
 
+    def _push_cache(self, alias=None, connection=None):
+        """
+         Overlay _cache.register using dictionary
+         Create a dictionary that contains the dbconnection and the api_module used
```
Comment (lucagiove) [inline src/MongoDBLibrary/mongo_connection_manager.py:25]: api_module was for postgresql

File: src/MongoDBLibrary/mongo_connection_manager.py [hunk 5]
```diff
@@ -25,26 +51,35 @@ def connect_to_mongodb(self, dbHost='localhost', dbPort=27017, dbMaxPoolSize=10,
         | Connect To MongoDB | foo.bar.org | ${27017} |
         | # Or for an authenticated connection, note addtion of "mongodb://" to host uri |
         | Connect To MongoDB | mongodb://admin:admin@foo.bar.org | ${27017} |
+        | # Or for an connection with db uri |
+        | Connect To MongoDB | mongodb://admin:admin@foo.bar.org/27017 |
 
+        Added new field alias
```
Comment (lucagiove) [inline src/MongoDBLibrary/mongo_connection_manager.py:57]: ? this is not a changelog but method documentation

**Pull request merged**